Automattic/wp-calypso
# BBE: add custom page type

Fixes https://github.com/Automattic/martech/issues/2165

## Proposed Changes

* Adds a new custom page type to the BBE flow.
* The custom page type should have a blank thumbnail.
* On the website content form, the title of the custom page should be editable.
* The user should be able to upload up to 8 images or videos.

### Screenshots

<img width="429" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/a158bdf4-fe6a-46fb-bb5e-cfd109044fad">


<img width="843" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/9cc5042e-16f5-4c8e-b0c7-58e705609b03">

<img width="814" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/3af213c8-1cd0-4c62-acee-c328598d2285">


https://github.com/Automattic/wp-calypso/assets/5436027/64f7cf54-59bd-4a3b-ac5a-016fb94e4156



## Testing Instructions



* Go to `/start/do-it-for-me` and proceed through the flow steps till you reach the page picker step.
* Confirm that you see the Custom Page type and the title, description, and thumbnail match the screenshot.
* Select the custom page and complete the purchase.

-----

* On the website content collection form, confirm that you can see a text box to edit the title of the page.
* The text box should be empty by default.
* Leave the title textbox blank and enter some content in the context text field. Click on Next.
* Confirm that you can see a validation error for the title box. 
* Also test the other form sections and confirm that form validation works correctly.

-----

* Enter some text in the title text field. Click on "Save Changes" or "Next".
* Reload the page. Confirm that the title you entered previously is visible in the textbox.

-----

* Confirm that you can upload both images and videos in the media upload section.

## Pre-merge Checklist
